src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -1268,34 +1430,38 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
         continue;
       }
 
-      const audioAdaptation = getSupportedAdaptations(period, "audio")[0];
-      trackStorePeriod.audio.storedSettings =
-        audioAdaptation === undefined
-          ? null
-          : {
-              adaptation: audioAdaptation,
-              switchingMode: this._defaultAudioTrackSwitchingMode,
-              lockedRepresentations: new SharedReference<IRepresentationsChoice | null>(
-                null,
-              ),
-            };
+      const audioAdaptation = getSupportedAdaptations(period, "audio")[0] as
+        | IAdaptationMetadata
+        | undefined;
+      if (audioAdaptation === undefined) {
+        trackStorePeriod.audio.storedSettings = null;
+        this.onNoPlayableRepresentation(period, "audio");
```

Comment:
/!\ I'm under the impression that this method may lead to `noPlayableTrack` events which communicate about Periods not yet advertised through `newAvailablePeriods` events, so the application is generally not supposed to know about them in the order of events we had until now :/